kube/services/revproxy/revproxy-service-elb.yaml

```diff
@@ -0,0 +1,20 @@
+kind: Service
+apiVersion: v1
+metadata:
+  name: revproxy-service-elb
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
+    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
+    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: $ARN
+    service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
+    # supported in k8s 1.9
+    service.beta.kubernetes.io/aws-load-balancer-ssl-negotiation-policy: "ELBSecurityPolicy-TLS-1-2-2017-01"
+spec:
+  selector:
+    app: revproxy
+  ports:
+    - protocol: TCP
+      port: 443
+      targetPort: 81
```

Comment:
who has port 81 open?

Reply:
So using SSL TCP on ELB breaks the http forwarded for headed, which means logging can't log the client IP. We fix this by using proxy protocol on the ELB. RevProxy Nginx then exposes port 81 which reads the proxy protocol + services requests. So the ELB -> revproxy nginx happens on port 81. Internal traffic still talks to revproxy on port 80 since it isn't sending the proxy protocol headers.